physical/raft/raft.go

```diff
@@ -209,7 +209,15 @@ type RaftBackend struct {
 // HookInvalidate implements physical.CacheInvalidationBackend.
 func (r *RaftBackend) HookInvalidate(hook physical.InvalidateFunc) {
 	r.fsm.hookInvalidate(func(key string) {
-		_, leaderId := r.raft.LeaderWithID()
+		r.l.RLock()
+		raft := r.raft
+		r.l.RUnlock()
+
+		if raft == nil {
+			return
+		}
+		_, leaderId := raft.LeaderWithID()
```

Comment:
@phil9909 🤔 Do we think it is safe to call `LeaderWithID()` if `raft` changes under us? Or should we hold the `r.l` lock through this entire thing? 

I don't really see any other callers of `LeaderWithID()` but for things like `State()`, they hold the lock throughout the call... 🤔

Reply:
The raft object itself has it's own lock. So yes, this should be safe.